scif/main/install.py

```diff
@@ -158,7 +158,9 @@ def install_commands(self, app, settings, config):
         # issue install commands
         cmd = "\n".join(config["appinstall"])
```

Comment:
I think multiple commands need to be `join`-ed by `&&`:
```suggestion
         cmd = " && ".join(config["appinstall"])
```

Otherwise it would proceed running commands after a failed command and the returned exit code would be the exit code of the last command (and not of the failed command).

Reply:
I don't think this would be the case (could you give an example that would do that?) Here is what I just tested:

```python
import os 
config = {"appinstall":["return 0", "return 1", "return 2"]}            
cmd = "\n".join(config["appinstall"])                                   
cmd                                                                     
'return 0\nreturn 1\nreturn 2'
retval = os.system(cmd)                                                 
retval                                                                  
0

config = {"appinstall":["echo 0", "echo 1", "echo 2"]}                  
cmd = "\n".join(config["appinstall"])                                  
retval = os.system(cmd)                                                
0
1
2


```

Reply:
You first test shows a slightly different but also wrong behavior: 

`retval` is still zero despite that `return 1` or `return 2` should lead to failure.

Please see the following example:

```python
import os

cmd = 'some_nonexisting_cmd\necho This should not be printed!'
retval = os.system(cmd)
if retval != 0:
    print("ERROR: Return value %s." % (retval))
else:
    print("SUCCESS")

print('-------------------------------')

cmd = 'some_nonexisting_cmd && echo Foo'
retval = os.system(cmd)
if retval != 0:
    print("ERROR: Return value %s." % (retval))
else:
    print("SUCCESS")
```
Output:
```
sh: some_nonexisting_cmd: command not found
This should not be printed!
SUCCESS
-------------------------------
sh: some_nonexisting_cmd: command not found
ERROR: Return value 32512.
```

Reply:
@bilke if the first test has the first line return 0, if this were in a script I would not expect it to continue to the next lines. The behavior I showed is expected for that. The second set of echos I did to show that we continue beyond the first line given that a return statement isn't used.

But your example is definitely an issue - we shouldn't have this behavior, and the reason isn't to add the && but that the bash shell isn't set to exit on error. Give this a try and let me know what you think:

```bash
cmd = 'set -e\nsome_nonexisting_cmd\necho This should not be printed!' 
retval = os.system(cmd)                                                                                             
sh: 2: some_nonexisting_cmd: not found
```
The retval is  32512 and the final line is not printed.

The issue with adding && is that some of those run scripts are already going to have them, and we shouldn't be enforcing / changing the actual script that the user has provided.

Reply:
I updated the PR to use strict mode, please take another look!

Reply:
@vsoch Sorry for the confusion! You are right the example with `return [something]` is completely misleading in this case!

I tested the strict mode, it works as I expected it. Thanks a lot!